include/dt-bindings/pinctrl/gd32f403r(k-i-g-e-c-b)xx-pinctrl.h

```diff
@@ -0,0 +1,907 @@
+
+/* SPI0_SCK */
+#define SPI0_SCK_PA5_INP_NORMP \
+	GD32_PINMUX_AFIO('A', 5, GPIO_IN, SPI0_NORMP)
+#define SPI0_SCK_PA5_OUT_NORMP \
+	GD32_PINMUX_AFIO('A', 5, ALTERNATE, SPI0_NORMP)
+#define SPI0_SCK_PB3_INP_RMP \
+	GD32_PINMUX_AFIO('B', 3, GPIO_IN, SPI0_RMP)
+#define SPI0_SCK_PB3_OUT_RMP \
+	GD32_PINMUX_AFIO('B', 3, ALTERNATE, SPI0_RMP)
+
+/* SPI1_MISO */
+#define SPI1_MISO_PB14_INP \
+	GD32_PINMUX_AFIO('B', 14, GPIO_IN, NORMP)
+#define SPI1_MISO_PB14_OUT \
+	GD32_PINMUX_AFIO('B', 14, ALTERNATE, NORMP)
+
+/* SPI1_MOSI */
+#define SPI1_MOSI_PB15_INP \
+	GD32_PINMUX_AFIO('B', 15, GPIO_IN, NORMP)
+#define SPI1_MOSI_PB15_OUT \
+	GD32_PINMUX_AFIO('B', 15, ALTERNATE, NORMP)
+
+/* SPI1_NSS */
+#define SPI1_NSS_PB12_INP \
+	GD32_PINMUX_AFIO('B', 12, GPIO_IN, NORMP)
+#define SPI1_NSS_PB12_OUT \
+	GD32_PINMUX_AFIO('B', 12, ALTERNATE, NORMP)
+
+/* SPI1_SCK */
+#define SPI1_SCK_PB13_INP \
+	GD32_PINMUX_AFIO('B', 13, GPIO_IN, NORMP)
+#define SPI1_SCK_PB13_OUT \
+	GD32_PINMUX_AFIO('B', 13, ALTERNATE, NORMP)
+
+/* SPI2_MISO */
+#define SPI2_MISO_PB4_INP_NORMP \
+	GD32_PINMUX_AFIO('B', 4, GPIO_IN, SPI2_NORMP)
+#define SPI2_MISO_PB4_OUT_NORMP \
+	GD32_PINMUX_AFIO('B', 4, ALTERNATE, SPI2_NORMP)
+#define SPI2_MISO_PC11_INP_RMP \
+	GD32_PINMUX_AFIO('C', 11, GPIO_IN, SPI2_RMP)
+#define SPI2_MISO_PC11_OUT_RMP \
+	GD32_PINMUX_AFIO('C', 11, ALTERNATE, SPI2_RMP)
+
+/* SPI2_MOSI */
+#define SPI2_MOSI_PB5_INP_NORMP \
+	GD32_PINMUX_AFIO('B', 5, GPIO_IN, SPI2_NORMP)
+#define SPI2_MOSI_PB5_OUT_NORMP \
+	GD32_PINMUX_AFIO('B', 5, ALTERNATE, SPI2_NORMP)
+#define SPI2_MOSI_PC12_INP_RMP \
+	GD32_PINMUX_AFIO('C', 12, GPIO_IN, SPI2_RMP)
+#define SPI2_MOSI_PC12_OUT_RMP \
+	GD32_PINMUX_AFIO('C', 12, ALTERNATE, SPI2_RMP)
+
+/* SPI2_NSS */
+#define SPI2_NSS_PA15_INP_NORMP \
+	GD32_PINMUX_AFIO('A', 15, GPIO_IN, SPI2_NORMP)
+#define SPI2_NSS_PA15_OUT_NORMP \
+	GD32_PINMUX_AFIO('A', 15, ALTERNATE, SPI2_NORMP)
+#define SPI2_NSS_PA4_INP_RMP \
+	GD32_PINMUX_AFIO('A', 4, GPIO_IN, SPI2_RMP)
+#define SPI2_NSS_PA4_OUT_RMP \
+	GD32_PINMUX_AFIO('A', 4, ALTERNATE, SPI2_RMP)
+
+/* SPI2_SCK */
+#define SPI2_SCK_PB3_INP_NORMP \
+	GD32_PINMUX_AFIO('B', 3, GPIO_IN, SPI2_NORMP)
+#define SPI2_SCK_PB3_OUT_NORMP \
+	GD32_PINMUX_AFIO('B', 3, ALTERNATE, SPI2_NORMP)
+#define SPI2_SCK_PC10_INP_RMP \
+	GD32_PINMUX_AFIO('C', 10, GPIO_IN, SPI2_RMP)
+#define SPI2_SCK_PC10_OUT_RMP \
+	GD32_PINMUX_AFIO('C', 10, ALTERNATE, SPI2_RMP)
+
+/* TAMPER_RTC */
+#define TAMPER_RTC_PC13 \
+	GD32_PINMUX_AFIO('C', 13, GPIO_IN, NORMP)
+
+/* TIMER0_BRKIN */
+#define TIMER0_BRKIN_PB12 \
+	GD32_PINMUX_AFIO('B', 12, GPIO_IN, NORMP)
+
+/* TIMER0_CH0 */
+#define TIMER0_CH0_PA8_INP_NORMP \
+	GD32_PINMUX_AFIO('A', 8, GPIO_IN, TIMER0_NORMP)
+#define TIMER0_CH0_PA8_OUT_NORMP \
+	GD32_PINMUX_AFIO('A', 8, ALTERNATE, TIMER0_NORMP)
+#define TIMER0_CH0_PA8_INP_PRMP \
+	GD32_PINMUX_AFIO('A', 8, GPIO_IN, TIMER0_PRMP)
+#define TIMER0_CH0_PA8_OUT_PRMP \
+	GD32_PINMUX_AFIO('A', 8, ALTERNATE, TIMER0_PRMP)
+#define TIMER0_CH0_PE9_INP_FRMP \
```

Comment:
GD32F403Rx(LQFP64) doesn't have the PE9 pin. AFIO remap have some package restriction, because the required pins doesn't exist on this package.

Reply:
nice catch, I've adjusted script to handle this case, now non-possible remaps will not be defined.

Reply:
This is very important @cameled. It was a doubt that I had and now it is clear.
However, we don't need address all this at this moment. I think we can start to work and make adjusts in future to cover exceptions.